snapcraft/project/_sanity_checks.py

```diff
@@ -45,6 +47,25 @@ def conduct_project_sanity_check(project: snapcraft.project.Project):
     if os.path.isdir(snap_dir_path):
         _check_snap_dir(snap_dir_path)
 
+    _migration_notice(project.info.base)
+
+
+def _migration_notice(base: Optional[str]) -> None:
+    # We only want to encourage developers to use bases if using the snapcraft snap.
+    if base is None and common.is_snap():
```

Comment:
Do we want to limit bases to **only** snap-packaged snapcraft? What about snapcraft running on Windows or macOS?

Reply:
You will install the snap in the VM and find yourself in this use case. I could add a comment if it makes it clearer. But at the end of the day you are going to be running the snap if running snapcraft from a native OSX and future Windows environment.